[annoy-o-mus] iP

annoy-o-mus opened the pull request.

annoy-o-mus committed:
feat: Add greetings

annoy-o-mus committed:
refactor: Make logo and section bar text final

annoy-o-mus committed:
refactor: Make greeting and goodbye as methods

annoy-o-mus committed:
docs: Update ASCII art logo

annoy-o-mus committed:
feat: Echo back user inputs

annoy-o-mus committed:
feat: Add task listing functionality

annoy-o-mus committed:
refactor: Shift goodbye invocation

annoy-o-mus committed:
refactor: Shift task related code to Task class

annoy-o-mus committed:
feat: Add task marking and unmarking

Introduces functionality to mark and unmark tasks and track the status
of tasks.

Task listing now reflects status of the task on top of description.

Basic error handling added to deal with `mark` and `unmark` commands
not having a compulsory task number supplied to it.

BenjoBurger commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,115 @@
+    }
+
+    /**
+     * Takes user inputs as commands and process them.
```
Good number of comments for each function/method and follows javadoc form

BenjoBurger commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,105 @@
+import java.util.ArrayList;
+
+public class Task {
+    private static final int MAX_TASKS = 128;
```
Good job ensuring variable names are correctly formatted to camelCase

BenjoBurger commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,105 @@
+     *
+     * @return The task description.
+     */
+    public String getTaskDescription() {
```
Method names are verbs with camelCase formatting.

BenjoBurger commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,115 @@
+
 public class Duke {
+    // ASCII Art Generated from: https://patorjk.com/software/taag/#p=display&f=Big&t=anonBot
+    private static final String LOGO = "                          ____        _   \n"
```
Good job for ensuring that the indentation for wrap around texts is 8 spaces

annoy-o-mus committed:
misc: Change name from Duke to AnonBot

Renames the program.

Also changes the test scripts and documenataion to reflect
the new changes.

annoy-o-mus committed:
refactor: Create Task Manager to manage Tasks

annoy-o-mus committed:
refactor: Decouple functionalities from AnonBot

Reintroduced functionalities as classes

* CommandManager - Processes the user's inputs as commands
* Parser - Handles Parsing related activities
* Status - Enum file containing status codes
* Ui - Functionalities relating to interating with user

annoy-o-mus committed:
refactor: Split User input into command and argument

annoy-o-mus committed:
fix: Add input checks

* Now Checks for empty user input
* User inputs now split into Command and optional argument
* Checks the argument associated with `mark` and `unmark` commands
to make sure it is correctly supplied

annoy-o-mus committed:
feat: Add support for todo, deadline, event

This currently does not support the specific date/time feature

annoy-o-mus committed:
feat: Add parsing support for deadline and event

annoy-o-mus committed:
feat: Print task upon creation

annoy-o-mus committed:
refactor: Modify printing of logo and section bar

* remove \n and replaceing it with the lineSeparator
* Reduce the number of calls to the SECTION_BAR text

annoy-o-mus committed:
refactor: Streamline code flow for CommandManager

annoy-o-mus committed:
refactor: Make sectionBar priinting a call to Ui

annoy-o-mus committed:
refactor: Shift parsing functionality to subtasks

Instead of having the TaskManager to handle the parsing of description
specific to deadline and event, we have those tasks handle it themselves
instead.

annoy-o-mus committed:
style: Add and remove whitespaces

annoy-o-mus committed:
refactor: Shift classes into packages

annoy-o-mus committed:
feat: Add InvalidCommandException and handler

The addition of AnonbotExceptionHandler will ensure that all Exceptions
that is defined in Anonbot will handle the printing of the error
messages.

annoy-o-mus committed:
misc: Cleanup unnecessary Status code

annoy-o-mus committed:
feat: Add IncompleteCommandException

annoy-o-mus committed:
feat: Exception handling: empty task description

annoy-o-mus committed:
style: Modify whitespaces

annoy-o-mus committed:
Merge branch 'branch-Level-5'

annoy-o-mus committed:
misc: Change folder structure

annoy-o-mus committed:
test: Add testing

annoy-o-mus committed:
feat: Add Exception to mark and unmark command

annoy-o-mus committed:
Merge branch 'branch-A-Exceptions'

annoy-o-mus committed:
feat: Add delete command

annoy-o-mus committed:
misc: Decouple active tasks from total tasks

In trying to keep the task ID unique (non-reusable), we need to keep
track of the new task ID to give and the number of active tasks
separately.

annoy-o-mus committed:
style: Modfy whitespaces

annoy-o-mus committed:
feat: Add saving of data

annoy-o-mus committed:
feat: Add loading of data

annoy-o-mus committed:
style: Add whitespaces

annoy-o-mus committed:
Merge branch 'branch-Level-6'

annoy-o-mus committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/anonbot/task/TaskManager.java

annoy-o-mus committed:
fix: Correct print sequence for adding of task

okkhoy commented on src/main/java/anonbot/misc/CommandManager.java:
```diff
@@ -0,0 +1,103 @@
+        try {
+            switch (command) {
+            case "exit":
+                // fallthrough
```
👍

okkhoy commented on src/main/java/anonbot/misc/CommandManager.java:
```diff
@@ -0,0 +1,103 @@
+        }
+    }
+
+    private static void processUnmarkCommand(String rawArgument) throws IncompleteCommandException {
```
all non-trivial methods need header comments

okkhoy commented on src/main/java/anonbot/misc/Parser.java:
```diff
@@ -0,0 +1,96 @@
+        String[] splitStringArray = rawDeadlineDescription.split(" /by ", 2);
+        parsedDeadlineSubstrings[0] = splitStringArray[0];
+
+        // We currently do not check if the deadline task has the right format (/by)
```
looks like a note-to-self comment.
if this is the intended behavior of the method, can document it in the header comment

okkhoy reviewed (commented):
In general, well written code 💯 
I have pointed out some minor issues with the comments; see if you can fix them before the submission

annoy-o-mus committed:
feat: Add find functionality

annoy-o-mus committed:
docs: Add javadoc for method

Added to this branch to test out the multiple PR functionality.

annoy-o-mus committed:
Merge pull request #1 from annoy-o-mus/branch-Level-9

Add Finding Functionality

annoy-o-mus committed:
Merge pull request #2 from annoy-o-mus/test-PartialJavaDoc

Add Partial JavaDocs

annoy-o-mus committed:
fix: Fix incorrect printing of total tasks

the `totalTasksCreated` was used previously as the delete functionality
was not introduced, and so the total number of tasks was the number
of the last item on the list. When delete was introduced, it was
decided that all new tasks created will have a unique Task ID. Hence,
there is now a need to separate check the number of active tasks in the
list.

annoy-o-mus committed:
Merge pull request #3 from annoy-o-mus/fix-incorrect-task-number

fix: Fix incorrect printing of total tasks

annoy-o-mus committed:
misc: Add error comment

This allows the user to consider the case where they might have
accidentally added leading spaces, resulting in no command being read.

annoy-o-mus committed:
refactor: Create new Command class

annoy-o-mus committed:
refactor: Introduce *ArgumentException

A more descriptive class that explicitly targets argument issues
that some of the commands may expect to have

annoy-o-mus committed:
refactor: Make delete command follow ArgumentException

annoy-o-mus committed:
refactor: Make tasks errors include syntax

annoy-o-mus committed:
feat: Make mark and unmark provide syntax

annoy-o-mus committed:
Merge pull request #4 from annoy-o-mus/fix-command-clarity

Fix command clarity

annoy-o-mus committed:
feat: Add help command

Added as a proper way to list all available commands.

annoy-o-mus committed:
Merge pull request #5 from annoy-o-mus/feat-help-command

feat: Add help command

Added as a proper way to list all available commands.

annoy-o-mus committed:
misc: Make program save after every change

Previously, it only saves on exit. With this new change, as long as
the list is changed, we save it.

annoy-o-mus committed:
Merge pull request #6 from annoy-o-mus/improve-save-condition

misc: Make program save after every change

annoy-o-mus committed:
misc: Update .gitignore

annoy-o-mus committed:
docs: Update comments in Parser class

As per feedback received from PR review in upstream

annoy-o-mus committed:
cleanup: Delete unused imports and code

annoy-o-mus committed:
misc: Adjust sementics for exception package

annoy-o-mus committed:
refactor: Streamline use of exception in import/export

annoy-o-mus committed:
feat: Add exception handling to initialisation

annoy-o-mus committed:
refactor: Allow AnonBotFile to be inherited

annoy-o-mus committed:
misc: Clean up Import-related functionality

annoy-o-mus committed:
refactor: Move execution status into Command class

annoy-o-mus committed:
misc: Clean Up export-related functionality

annoy-o-mus committed:
misc: Add handling for potential null tasks

annoy-o-mus committed:
misc: Change function name for Command class

annoy-o-mus committed:
refactor: Shift setter and getters

annoy-o-mus committed:
misc: Update text-ui-test

annoy-o-mus committed:
style: whitespces

annoy-o-mus committed:
Merge pull request #7 from annoy-o-mus/branch-A-MoreOOP

Branch A-MoreOOP

annoy-o-mus committed:
docs: Update JavaDoc for data storage

annoy-o-mus committed:
docs: Update JavaDoc for exception

annoy-o-mus committed:
docs: Update JavaDoc for commands

annoy-o-mus committed:
docs: Update JavaDoc for parsers

annoy-o-mus committed:
docs: Update JavaDoc for anonbot

annoy-o-mus committed:
docs: Update JavaDoc for tasks

annoy-o-mus committed:
docs: update README

annoy-o-mus committed:
Merge pull request #8 from annoy-o-mus/branch-A-JavaDoc

Branch A-JavaDoc

annoy-o-mus committed:
docs: Update User Guide to acknowledge source

annoy-o-mus committed:
docs: Add section links to document outline

annoy-o-mus committed:
doc: Fix broken link for feature

annoy-o-mus committed:
docs update saving and editing the data file

annoy-o-mus committed:
docs: remove HTML tags

annoy-o-mus committed:
docs: faq

annoy-o-mus committed:
docs: update help command

annoy-o-mus committed:
docs: Update task creation related commands

annoy-o-mus committed:
docs: Update list, find, exit commands

annoy-o-mus committed:
docs: Update mark, unmark, delete commands

annoy-o-mus committed:
docs: Fix code block issue

Suspects that the formatting screws up if the code block happens
right after the list symbol. Tries introducing blank lines to see
if it resolves the issue.

annoy-o-mus committed:
docs: Remove command summary support

annoy-o-mus committed:
Merge pull request #9 from annoy-o-mus/branch-A-UserGuide

Branch A-UserGuide

annoy-o-mus committed:
docs: fix wrong internal link

annoy-o-mus committed:
docs: minor punctuation and link changes

annoy-o-mus committed:
docs: Fix some javadoc grammar

annoy-o-mus committed:
Merge pull request #10 from annoy-o-mus/final-fixes

Final fixes
